pkg/resource/resource.go

```diff
@@ -249,6 +201,26 @@ func (a *Attributes) GetFloat64(path string) *float64 {
 	return &v
 }
 
+func (a *Attributes) GetMap(path string) map[string]interface{} {
```

Comment:
This should be unexported or done directly in `GetStringMap`.

BTW make a full copy of the map looks heavy for a simple type cast and I don't get why we don't do this :

![image](https://user-images.githubusercontent.com/6154987/120344117-95c08700-c2f9-11eb-8133-ee8f33ec9e0b.png)

Reply:
I added both the `GetMap` and `GetMapString` methods to have more method utilities in our pocket. And it totally make sense to only have the GetMap and cast the value at the end, didn't think of it.

Reply:
this was supposed to be one of the getter/autocast function like GetSlice, GetString, ...
And yeah I think we could replace the getstringmap you're right